subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/mvnsettings/DefaultLocalMavenRepositoryLocator.java

```diff
@@ -48,7 +48,12 @@ public File getLocalMavenRepository() throws CannotLocateLocalMavenRepositoryExc
         try {
             String repoPath = parseLocalRepoPathFromMavenSettings();
             if (repoPath != null) {
-                return new File(resolvePlaceholders(repoPath.trim()));
+                File file = new File(resolvePlaceholders(repoPath.trim()));
+                if (isDriveRelativeWindowsPath(file)) {
+                    return file.getAbsoluteFile();
```

Comment:
❌ This will interpret the path relative to the current working directory. Instead, it needs to interpret it relative to the settings.xml

Reply:
Well, I don't think so. When `isDriveRelativeWindowsPath(file)` return `true`, it means that the `localRepository` element is something like `<localRepository>/some-path</localRepository>` on Windows. Where this path is relative to is totally defined by Maven. And Maven decides it should be relative to drive root here: https://github.com/apache/maven/blob/master/maven-settings-builder/src/main/java/org/apache/maven/settings/building/DefaultSettingsBuilder.java#L106-L115

And we should be consistent with Maven.

Reply:
And maybe you're a bit confused something here:

on Windows, the following is true:

```
new File('/some-path').isAbsolute()==false
new File('/some-path').equals(new File('C:\\some-path'))==true
new File('\\some-path').isAbsolute()==false
new File('\\some-path').equals(new File('C:\\some-path'))==true
```

Reply:
Ah ,so Maven actually relies on this broken behavior on Windows, funny :) In that case I take back my comment above.